tasks/main.yml

```diff
@@ -1,32 +1,48 @@
 ---
-- name: Ensure /root/.aws/ is there
+- name: Ensure /{{ aws_cwa_local_user }}/.aws/ is there
   file:
-    path: '/root/.aws/'
+    path: "/{{ aws_cwa_local_user }}/.aws/"
     state: 'directory'
     mode: '0700'
-    owner: 'root'
-    group: 'root'
+    owner: "{{ aws_cwa_local_user }}"
+    group: "{{ aws_cwa_local_user }}"
+  when: aws_cwa_on_prem
 
-- name: Add AWS CWA profile to /root/.aws/config
+- name: Remove /{{ aws_cwa_local_user }}.aws/config
+  file:
+    path: "/{{ aws_cwa_local_user }}/.aws/config"
+    state: 'absent'
+  notify: restart amazon-cloudwatch-agent
+  when: not aws_cwa_on_prem
+
+- name: Remove /{{ aws_cwa_local_user }}.aws/credentials
+  file:
+    path: "/{{ aws_cwa_local_user }}/.aws/credentials"
+    state: 'absent'
+  notify: restart amazon-cloudwatch-agent
+  when: not aws_cwa_on_prem
+
+- name: Add AWS CWA profile to /{{ aws_cwa_local_user }}/.aws/config
   blockinfile:
-    path: '/root/.aws/config'
+    path: "/{{ aws_cwa_local_user }}/.aws/config"
     create: 'yes'
-    owner: 'root'
-    group: 'root'
+    owner: "{{ aws_cwa_local_user }}"
+    group: "{{ aws_cwa_local_user }}"
     mode: 0600
     marker: "# {mark} ansible aws-cwa cfg"
     block: |
       [profile AmazonCloudWatchAgent]
       output = json
       region = {{aws_cwa_region|default('us-east-1')}}
   notify: restart amazon-cloudwatch-agent
+  when: aws_cwa_on_prem
 
-- name: Add AWS CWA profile to /root/.aws/credentials
+- name: Add AWS CWA profile to /{{ aws_cwa_local_user }}/.aws/credentials
   blockinfile:
-    path: '/root/.aws/credentials'
+    path: "/{{ aws_cwa_local_user }}/.aws/credentials"
```

Comment:
That is only valid for `root` other users would be `/home/{{ aws_cwa_local_user }}`